wp-tevko-responsive-images.php

```diff
@@ -357,69 +358,71 @@ function tevkori_filter_attachment_image_attributes( $attr, $attachment, $size )
 add_filter( 'wp_get_attachment_image_attributes', 'tevkori_filter_attachment_image_attributes', 0, 3 );
 
 /**
- * Disable the editor size constraint applied for images in TinyMCE.
+ * Filter to add php-respimg as an image editor.
  *
- * @return array A width & height array so large it shouldn't constrain reasonable images.
- */
-function tevkori_editor_image_size() {
-	return array( 99999, 99999 );
-}
-
-/**
- * Load admin scripts.
+ * @since 2.3.0
  *
- * @param string $hook Admin page file name.
- */
-function tevkori_load_admin_scripts( $hook ) {
-	if ( $hook == 'post.php' || $hook == 'post-new.php' ) {
-		wp_enqueue_script( 'wp-tevko-responsive-images', plugin_dir_url( __FILE__ ) . 'js/wp-tevko-responsive-images.js', array( 'wp-backbone' ), '2.0.0', true );
+ * @return array Editors.
+ **/
+function tevkori_wp_image_editors( $editors ) {
+	if ( current_theme_supports( 'advanced-image-compression' ) ) {
+		array_unshift( $editors, 'WP_Image_Editor_Respimg' );
 	}
-}
-add_action( 'admin_enqueue_scripts', 'tevkori_load_admin_scripts' );
 
+	return $editors;
+}
+add_filter( 'wp_image_editors', 'tevkori_wp_image_editors' );
 
 /**
- * Filter for the_content to replace data-size attributes with size attributes.
+ * Convert multiple attachement URLs to thier ID.
  *
- * @since 2.2.0
+ * @since 3.0
  *
- * @param string $content The raw post content to be filtered.
- */
-function tevkori_filter_content_sizes( $content ) {
-	$images = '/(<img\s.*?)data-sizes="([^"]+)"/i';
-	$sizes = '${2}';
+ * @return array IDs.
+ **/
+function tevkori_attachment_urls_to_loop( $urls ) {
+	global $wpdb; 
 
-	return preg_replace( $images, '${1}sizes="' . $sizes . '"', $content );
-}
-add_filter( 'the_content', 'tevkori_filter_content_sizes' );
+	if ( is_string( $urls ) ) {
+		return tevkori_attachment_urls_to_loop( array( $urls ) );
+	}
+	
+	if ( ! is_array( $urls ) ) {
+		$urls = array();
+	}
 
+	$dir = wp_upload_dir();
+	$paths = $urls;
 
-/**
- * Ajax handler for updating the srcset when an image is changed in the editor.
- *
- * @since 2.3.0
- *
- * @return string A sourcest value.
- */
-function tevkori_ajax_srcset() {
+	$site_url = parse_url( $dir['url'] );
 
-	// Bail early if no post ID is passed.
-	if ( empty( $_POST['postID'] ) ) {
-		return;
-	}
+	foreach ( $paths as $k => $path ) {
+		$image_path = parse_url( $path );
```

Comment:
Here we could probably add the check if the file path matches with the upload dir and skip if not, as suggested by @joemcgill in https://github.com/ResponsiveImagesCG/wp-tevko-responsive-images/pull/144#commitcomment-13027001